Add kobuk-gfx changes onto upstream and update

mckees opened the pull request:
This takes our changes and merges them with the current state of the Ubuntu archive to allow for an easier review process

mckees committed:
Add kobuk-gfx changes onto upstream and update

androniychuk commented on debian/control:
```diff
@@ -1,22 +1,11 @@
-Source: libvpl-disp
+Source: onevpl
```
I'm not sure I like the name change

mckees replied:
Ya, that wasn’t me. It's already renamed in the Ubuntu archive source package as well as in Debian upstream

androniychuk replied:
onevpl can be confused with... vpl gpu runtime
or the older vpl, which was clumped together
maybe we can change it upstream?

mckees replied:
If that's a priority to change a source package name, I can ask about it more. But there is a history of using this name already established (and most users will never know the difference), I'm inclined to not use our social currency with maintainers on this

androniychuk replied:
I guess we can keep this is as onevpl, if we have to fight our battles

tjaalton replied:
well it was called onevpl, and then Intel decided to drop that branding ;)

the package source might get renamed to just libvpl at some point, but we'll see

tjaalton replied:
oh well, since the next upload will add libvpl-examples and would go through the NEW queue due to that, I've renamed the source now

mckees replied:
@tjaalton Did you rename it to libvpl or libvpl-disp like we have here?

mckees replied:
I went with libvpl-disp since Pavel made a case before that it was more descriptive to reference the dispatcher

tjaalton replied:
the upstream project name is libvpl, so that's what the debian (source) package was renamed to

androniychuk commented on debian/libvpl-dev.install:
```diff
@@ -1,9 +1,10 @@
-etc/vpl
```
where did this go?

mckees replied:
There's nothing in /etc to install

androniychuk replied:
because this is sending it to a diff location...
DVPL_INSTALL_ENVDIR=/usr/share/vpl/env \

But why is this necessary?

mckees replied:
It's Timo's solution, so I'm not totally sure. I do know that he originally did not specify this install env dir, but he added it here calling it a fix:
https://salsa.debian.org/debian/onevpl/-/commit/ff25b6c95e5a21077e64df879bbfed4ea2e3a281

It seems pretty intentional to me

androniychuk replied:
I'm not sure we can move away from /etc/vpl and still have the 1api stack working

mckees replied:
If we have confirmation from onevpl devs that this breaks things, I'm sure we can get it changed. Especially if we have a test example

tjaalton replied:
the only thing that would end up in there is a vars.sh script, which IMO wouldn't belong under /etc anyway

androniychuk commented on debian/control:
```diff
@@ -1,22 +1,11 @@
- libxcb-dri3-dev,
- libxcb-present-dev,
- pkg-config,
- wayland-protocols
```
i will need to test this, because if you build it without certain dependencies installed, you get different behavior

mckees replied:
If that's the case, can we get a rationale for why the build doesn't fail without those deps?

androniychuk replied:
because u can still use the library but just without the additional features

mckees replied:
Ouch. Would be ideal for those features be enabled/disabled with build flags so that it doesn't fail silently. Please let me know if that's the case though, and we will add the build deps back in, and I'll file a bug

androniychuk replied:
but features are not mandatory

mckees replied:
Sure, and that's why having them turned off or on by default would be ideal IMO. Then the user can consciously pick and choose features. But it's pretty unpredictable to have features silently turned on at build time based on what's installed on the build machine. 

But ya, if that's how it works, that's how it works. We can re-add the deleted deps, and I'll file a bug against the libvpl repo to bring up this issue

androniychuk replied:
I would turn examples ON and include libva-dev in the build-requires

androniychuk commented on debian/libvpl-dev.install:
```diff
@@ -1,9 +1,10 @@
-etc/vpl
-usr/include/vpl/*
-usr/lib/*/cmake/vpl/*
+usr/include
```
shouldn't this be more specific
usr/include/vpl/*  ?

mckees replied:
They are functionally the same thing since vpl is the only directory in include/, so I don't see the use in making a change to the upstream version

androniychuk replied:
that's true, I guess I see my frame of reference as what we have now....first and not upstream

androniychuk replied:
I would keep the install location as etc/vpl
I don't know how 1api behaves with a different location

tjaalton replied:
it's a config flag, the purpose is to allow changing the location, so I don't see why it would break anything

androniychuk commented on debian/rules:
```diff
@@ -1,10 +1,17 @@
 
+override_dh_auto_install:
+	dh_auto_install
+	rm -rf debian/tmp/usr/share/doc/oneVPL/license.txt \
```
why remove it?
just keep it

mckees replied:
Because the user is uninstalling the package at this stage, so this takes a "leave no trace" mindset

mckees replied:
Looking into this but suspect it's probably cleaned up because we already have a copyright file which defines the licenses

androniychuk replied:
does the license cover the same things? Like for example does this license cover the examples?

mckees replied:
Ya, and it has a couple cases we missed

mckees replied:
@tjaalton while you're here, what was the rationale for explicitly removing the license file here?

tjaalton replied:
seems like the removals here are just old cruft that can be dropped, so the override can be dropped too

I'll do that in the next debian upload

mckees replied:
Great, thanks. Let's just leave it as-is here then, and the upstream changes can take care of it

androniychuk commented:
I'm wondering what the impact is when installing the files into a different location
```
14:37:40,971 INFO  - -- Install:
14:37:40,971 INFO  - --   CMAKE_INSTALL_PREFIX            : /usr
14:37:40,971 INFO  - --   CMAKE_INSTALL_FULL_BINDIR       : /usr/bin
14:37:40,971 INFO  - --   CMAKE_INSTALL_FULL_INCLUDEDIR   : /usr/include
14:37:40,971 INFO  - --   CMAKE_INSTALL_FULL_LIBDIR       : /usr/lib/x86_64-linux-gnu
14:37:40,971 INFO  - --   CMAKE_INSTALL_FULL_DOCDIR       : /usr/share/doc/vpl
14:37:40,971 INFO  - --   VPL_INSTALL_FULL_PKGCONFIGDIR   : /usr/lib/x86_64-linux-gnu/pkgconfig
14:37:40,971 INFO  - --   VPL_INSTALL_FULL_CMAKEDIR       : /usr/lib/x86_64-linux-gnu/cmake
14:37:40,971 INFO  - --   VPL_INSTALL_FULL_ENVDIR         : /etc/vpl
14:37:40,972 INFO  - --   VPL_INSTALL_FULL_MODFILEDIR     : /etc/modulefiles
14:37:40,972 INFO  - --   VPL_INSTALL_FULL_EXAMPLEDIR     : /usr/share/vpl/examples
14:37:40,972 INFO  - --   VPL_INSTALL_FULL_LICENSEDIR     : /usr/share/vpl/licensing
14:37:40,972 INFO  - --   INSTALL_DEV                     : ON
14:37:40,972 INFO  - --   INSTALL_LIB                     : ON
14:37:40,972 INFO  - -- Build:
14:37:40,972 INFO  - --   BUILD_SHARED_LIBS               : ON
14:37:40,972 INFO  - --   BUILD_TESTS                     : OFF
14:37:40,972 INFO  - --   BUILD_EXAMPLES                  : ON
14:37:40,972 INFO  - --   BUILD_EXPERIMENTAL              : ON
14:37:40,972 INFO  - --   ENABLE_LIBDIR_IN_RUNTIME_SEARCH : OFF
```
also, I noticed that we build with Examples ON. I would like to keep that.

androniychuk commented:
when turning examples ON, then I get build failures
see below
```
-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api1x_core/legacy-decode/CMakeLists.txt:65 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build legacy-decode without libva
  support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api1x_core/legacy-vpp/CMakeLists.txt:65 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build legacy-vpp without libva
  support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api1x_core/legacy-encode/CMakeLists.txt:65 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build legacy-encode without libva
  support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api2x/hello-decode/CMakeLists.txt:69 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build hello-decode without libva
  support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api2x/hello-sharing-vaapi/CMakeLists.txt:72 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build hello-sharing-vaapi-export
  without libva support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api2x/hello-sharing-vaapi/CMakeLists.txt:72 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build hello-sharing-vaapi-import
  without libva support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api2x/hello-decvpp/CMakeLists.txt:69 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build hello-decvpp without libva
  support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api2x/hello-encode/CMakeLists.txt:68 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build hello-encode without libva
  support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api2x/hello-encode-jpeg/CMakeLists.txt:68 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build hello-encode-jpeg without
  libva support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api2x/hello-transcode/CMakeLists.txt:69 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build hello-transcode without
  libva support


-- Checking for module 'libva>=1.2'
--   Package 'libva', required by 'virtual:world', not found
-- Checking for module 'libva-drm>=1.2'
--   Package 'libva-drm', required by 'virtual:world', not found
CMake Error at examples/api2x/hello-vpp/CMakeLists.txt:78 (message):
  libva not found: set LIBVA_SUPPORT=OFF to build hello-vpp without libva
  support
  ```
so at least libva needs to be on that list

mckees committed:
Fix install issues from both packages containing license

androniychuk commented:
@mckees  is the purpose of this PR to get an updated pkg in the kobuk PPA, or to prep the pkgs for the noble update in Feb ?

mckees commented:
> @mckees is the purpose of this PR to get an updated pkg in the kobuk PPA, or to prep the pkgs for the noble update in Feb ?

Both. We can keep it from diverging from upstream while also getting it ready for SRU submission. This is the base I will use to create the debdiffs

mckees committed:
Add upstream metadata from the archive

androniychuk commented:
Our pkgs in the PPA have the examples enabled
we should not regress this part

as for the other changes, I'm ok to implement.

the main ones:
1. location change of binaries
2. build-deps update (please keep libva-dev)

mckees committed:
Fix build error with examples enabled

mckees committed:
Separate libvpl-examples into a separate package

mckees commented:
> Our pkgs in the PPA have the examples enabled we should not regress this part
> 

They were never removed from the build. It just looked like it because I added a backslash to the end of the line

> as for the other changes, I'm ok to implement.
> 
> the main ones:
> 
> 1. location change of binaries

Did we have confirmation that it breaks things? Otherwise, to Timo's point, it's a config flag. It's purpose is to divert the location

> 2. build-deps update (please keep libva-dev)

Thanks for catching that. Typical case of "I had it installed, so nothing complained when I removed the dep"

I also added a few more patches, one of which adds libvpl-examples

androniychuk commented:
> > Our pkgs in the PPA have the examples enabled we should not regress this part
> 
> They were never removed from the build. It just looked like it because I added a backslash to the end of the line
> 
> > as for the other changes, I'm ok to implement.
> > the main ones:
> > 
> > 1. location change of binaries
> 
> Did we have confirmation that it breaks things? Otherwise, to Timo's point, it's a config flag. It's purpose is to divert the location
> 
> > 2. build-deps update (please keep libva-dev)
> 
> Thanks for catching that. Typical case of "I had it installed, so nothing complained when I removed the dep"
> 
> I also added a few more patches, one of which adds libvpl-examples

vpl team said it shouldn't regress 1api. 
that's good enough for me

androniychuk commented on debian/changelog:
```diff
@@ -1,32 +1,21 @@
-libvpl-disp (1:2.14.0-0ubuntu1~24.04~ppa1) noble; urgency=medium
-
+onevpl (1:2.14.0-1) unstable; urgency=medium
```
noble?

mckees replied:
Changed it in the latest commit

mckees committed:
Update changelog with a better description of changes

androniychuk commented on debian/changelog:
```diff
@@ -1,32 +1,23 @@
-libvpl-disp (1:2.14.0-0ubuntu1~24.04~ppa1) noble; urgency=medium
-
+libvpl-disp (1:2.14.0-1) noble; urgency=medium
```
u don't like the "ubuntu1" stuff? I don't either, I'd rather remove it completely

mckees replied:
Actually that was a mistake. The ubuntu1 stuff is needed if it's an ubuntu change:
https://canonical-ubuntu-packaging-guide.readthedocs-hosted.com/en/latest/reference/debian-dir-overview/#the-changelog-file

androniychuk replied:
dang, almost got away with it

androniychuk commented on debian/control:
```diff
@@ -25,39 +15,54 @@ Rules-Requires-Root: no
  .
  This package includes header files and examples for development.
+
+Package: libvpl-examples
```
whats the rational behind, not stuffing this into the dev pkg ?

mckees replied:
So we don't include a bunch of binaries and media samples in a package unless people care about using it for that purpose

androniychuk commented:
ok...looks like 2 more things needed

source pkg name change to what Timo will have upstream
drop the unnecessary overrides

And looks like we are good to merge

mckees committed:
Change source package name and unneeded install locations

mckees commented:
> ok...looks like 2 more things needed
> 
> source pkg name change to what Timo will have upstream 

Done

> drop the unnecessary overrides

I assume you mean the build flags. Let's give it a swing and see if it gets through. New commit pushed
